client/my-sites/checkout/checkout/composite-checkout-container.js

```diff
@@ -37,6 +37,11 @@ async function sendStripeTransaction() {
 }
 
 const stripeMethod = createStripeMethod( {
+	getSiteId: () => select( 'wpcom' )?.getSiteId?.(),
```

Comment:
Good question! It would be passed along to the server and the transaction would (hopefully!) fail. However, the failure message could possibly be unhelpful and it might be worth putting some safeguards in the code to fail earlier. On the other hand that's tricky to do because when the form initially loads the site Id won't be available (it has to be passed into the store which will take at least one render cycle).

Reply:
Good question! It would be passed along to the server and the transaction would fail. The error message might be poor though so it is probably worth putting some validation in before submission.